src/zenml/integrations/mlflow/mlflow_step_decorator.py

```diff
@@ -152,20 +149,18 @@ def wrapper(*args: Any, **kwargs: Any) -> Any:  # noqa
                 func.__name__,
             )
             # Create and activate the global MLflow environment
-            global_mlflow_env = MLFlowEnvironment()
             step_env = Environment().step_environment
             experiment = experiment_name or step_env.pipeline_name
             step_mlflow_env = MLFlowStepEnvironment(
                 experiment_name=experiment, run_name=step_env.pipeline_run_id
             )
-            with global_mlflow_env:
-                with step_mlflow_env:
+            with step_mlflow_env:
 
-                    # should never happen, but just in case
-                    assert step_mlflow_env.mlflow_run is not None
+                # should never happen, but just in case
+                assert step_mlflow_env.mlflow_run is not None
 
-                    with step_mlflow_env.mlflow_run:
-                        return func(*args, **kwargs)
+                with step_mlflow_env.mlflow_run:
+                    return func(*args, **kwargs)
```

Comment:
It's great getting rid of the global mlflow environment, I always felt it was out of place. Now that we have an experiment tracking stack component, you might also think about generalizing what the step mlflow environment does: setting up the step-level session configuration for a pipeline. It could be something that is also done in the experiment tracking component instead of the step mlflow environment. I could also be something that, if part of the generic experiment tracker component abstraction, is implemented similarly for the WAND component.

Reply:
I can see storing the `experiment_name` there and have it create the mlflow run, would we then inside the mlflow step decorator call the step function using a `with active_stack.experiment_tracker.active_mlflow_run:` or something similar?

Reply:
I can see storing the `experiment_name` there and have it create the mlflow run, would we then inside the mlflow step decorator call the step function using a `with active_stack.experiment_tracker.active_mlflow_run:` or something similar?

Reply:
I just got rid of the mlflow step environment with a contextmanager on the mlflow/wandb experiment trackers. Not sure if that's what you had in mind @stefannica, happy for any feedback to improve it